anomaly_detector/anomaly_detector.py

```diff
@@ -149,86 +154,53 @@ def infer(self, false_positives=None):
         _LOGGER.info("threshold for anomaly is of %f" % threshold)
         _LOGGER.info("Models loaded, running %d infer loops" % self.config.INFER_LOOPS)
 
-        infer_loops = 0
-        while infer_loops < self.config.INFER_LOOPS:
-            INFERENCE_COUNT.inc()
-            then = time.time()
-            now = datetime.datetime.now()
+        INFERENCE_COUNT.inc()
+        then = time.time()
+        f = []
+        _LOGGER.info("Max anomaly score: %f" % max(dist))
+        for i in range(len(data)):
+            _LOGGER.debug("Updating entry %d - dist: %f; mean: %f" % (i, dist[i], mean))
+
+            # TODO: This needs to be more general,
+            #       only works for ES incoming logs right now.
+            s = json_logs[i]
+            s["predict_id"] = str(uuid.uuid4())
+            s["anomaly_score"] = dist[i]
```

Comment:
@zmhassan This approach is a little bit confusing to me. I see that you are setting the anomaly score from the `dist[ ]` list. But you are providing that list as an argument to the `infer()` function. The way I think about it is like this: computing the `dist[ ]` values and then checking whether they are higher or lower than the threshold is the inference step. And I don't see where those distance values are being calculated for new data. I only see these values being calculated in the the `train()` function, so its unclear to me where the inference on new data is happening. Let me know if I am mistaken, but this reads to me like training and inference is being performed on the same data. Let me know if I've overlooked something : ) thanks.

Reply:
That is correct. I am using the same data that was used to train to do the inference. 

https://github.com/AICoE/log-anomaly-detector/blob/4e8834da9f96409d01fee6f9b4cdf762f23e490d/anomaly_detector/anomaly_detector.py#L221-L238

In the run function you'll see I pass that in and we also pass in the false positives as well.

Reply:
```             self.infer(false_positives=false_positives, dist=dist, data=data, json_logs=json_logs) ```